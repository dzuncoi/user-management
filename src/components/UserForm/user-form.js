import React from 'react'
import PropTypes from 'prop-types'
import { Form, Field } from 'formik'
import { Row, Col } from 'antd'

import TextInput from '../Input/input'
import Select, { Option } from '../Select/select'

import {
  email as emailValidation,
  required as requiredValidation,
} from '../../helpers/validations/validations'

import styles from './user-form.module.scss'

const UserForm = props => (
  <div className={styles.container}>
    <Form>
      <Row type="flex" gutter={16} className={styles.formRow}>
        <Col span={24} md={{ span: 12 }}>
          <Field
            type="name"
            name="firstName"
            id="firstName"
            placeholder="First Name"
            component={TextInput}
            validate={requiredValidation('First name is required!')}
          />
        </Col>
        <Col span={24} md={{ span: 12 }}>
          <Field
            type="name"
            name="lastName"
            id="lastName"
            placeholder="Last Name"
            component={TextInput}
            validate={requiredValidation('Last name is required!')}
          />
        </Col>
      </Row>
      <Row type="flex" gutter={16} className={styles.formRow}>
        <Col span={24} md={{ span: 12 }}>
          <Field
            type="email"
            name="email"
            id="email"
            placeholder="Email"
            component={TextInput}
            validate={emailValidation}
          />
        </Col>
        <Col span={24} md={{ span: 12 }}>
          <Field
            type="text"
            name="groups"
            id="groups"
            component={Select}
            placeholder="Select group(s)"
            mode="multiple"
            optionFilterProp="children"
          >
            {props.groups.map(group => (
              <Option key={group.id} value={group.id}>{group.name}</Option>
            ))}
          </Field>
        </Col>
      </Row>
    </Form>
  </div>
)

UserForm.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
  })),
}

UserForm.defaultProps = {
  groups: [],
}

export default UserForm
