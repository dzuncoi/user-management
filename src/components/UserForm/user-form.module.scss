.container {
  padding: 0 16px;
}

.formRow {
  margin-bottom: 16px;
}
