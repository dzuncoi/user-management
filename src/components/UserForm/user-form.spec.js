import React from 'react'

import UserForm from './user-form'

jest.mock('formik', () => ({
  Form: () => 'Form',
  Field: () => 'Field',
}))
jest.mock('antd', () => ({
  Row: () => 'Row',
  Col: () => 'Col',
}))
jest.mock('../Input/input', () => 'TextInput')
jest.mock('../Select/select', () => ({
  default: () => 'Select',
  Option: () => 'Option',
}))

describe('user-form', () => {
  let props

  beforeEach(() => {
    props = {
      groups: [{
        id: '1',
        name: 'group 1',
      }],
    }
  })

  it('renders', () => {
    expect(shallow(<UserForm {...props} />)).toMatchSnapshot()
  })
})
