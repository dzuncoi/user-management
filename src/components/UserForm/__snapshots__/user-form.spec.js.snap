// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user-form renders 1`] = `
<div
  className="container"
>
  <Form>
    <Row
      className="formRow"
      gutter={16}
      type="flex"
    >
      <Col
        md={
          Object {
            "span": 12,
          }
        }
        span={24}
      >
        <Field
          component="TextInput"
          id="firstName"
          name="firstName"
          placeholder="First Name"
          type="name"
          validate={[Function]}
        />
      </Col>
      <Col
        md={
          Object {
            "span": 12,
          }
        }
        span={24}
      >
        <Field
          component="TextInput"
          id="lastName"
          name="lastName"
          placeholder="Last Name"
          type="name"
          validate={[Function]}
        />
      </Col>
    </Row>
    <Row
      className="formRow"
      gutter={16}
      type="flex"
    >
      <Col
        md={
          Object {
            "span": 12,
          }
        }
        span={24}
      >
        <Field
          component="TextInput"
          id="email"
          name="email"
          placeholder="Email"
          type="email"
          validate={[Function]}
        />
      </Col>
      <Col
        md={
          Object {
            "span": 12,
          }
        }
        span={24}
      >
        <Field
          component={
            Object {
              "Option": [Function],
              "default": [Function],
            }
          }
          id="groups"
          mode="multiple"
          name="groups"
          optionFilterProp="children"
          placeholder="Select group(s)"
          type="text"
        >
          <Option
            key="1"
            value="1"
          >
            group 1
          </Option>
        </Field>
      </Col>
    </Row>
  </Form>
</div>
`;
